algo/src/test/java/com/hartwig/serve/sources/ckb/CkbKnownEventsExtractorTest.java

```diff
@@ -0,0 +1,131 @@
+package com.hartwig.serve.sources.ckb;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import com.hartwig.serve.common.classification.EventType;
+import com.hartwig.serve.datamodel.molecular.KnownEvents;
+import com.hartwig.serve.datamodel.molecular.MutationType;
+import com.hartwig.serve.datamodel.molecular.common.ProteinEffect;
+import com.hartwig.serve.datamodel.molecular.fusion.FusionTestFactory;
+import com.hartwig.serve.datamodel.molecular.gene.GeneEvent;
+import com.hartwig.serve.datamodel.molecular.hotspot.HotspotTestFactory;
+import com.hartwig.serve.datamodel.molecular.range.RangeTestFactory;
+import com.hartwig.serve.extraction.ImmutableEventExtractorOutput;
+import com.hartwig.serve.extraction.codon.ImmutableCodonAnnotation;
+import com.hartwig.serve.extraction.exon.ImmutableExonAnnotation;
+import com.hartwig.serve.extraction.gene.ImmutableGeneAnnotationImpl;
+
+import org.apache.commons.compress.utils.Lists;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.Test;
+
+public class CkbKnownEventsExtractorTest {
+
+    @Test
+    public void canFilterUnknownProteinEffectFromKnownEvents() {
+        List<ExtractedEvent> extractedEvents = Lists.newArrayList();
+        extractedEvents.add(createCodon("BRAF", "V600", "gain of function"));
+        extractedEvents.add(createCodon("TP53", "R175", "unknown"));
+        extractedEvents.add(createExon("BRAF", "exon 1 deletion", "loss of function"));
+        extractedEvents.add(createExon("APC", "exon 3 deletion", "unknown"));
+        extractedEvents.add(createFusion("ALK", "EML4", "gain of function"));
+        extractedEvents.add(createFusion("APC", "BRAF", "unknown"));
+        extractedEvents.add(createHotspot("BRAF", "V600E", "gain of function"));
+        extractedEvents.add(createHotspot("TP53", "R175H", "unknown"));
+        extractedEvents.add(createAmplification("MET", "gain of function"));
+        extractedEvents.add(createAmplification("KIT", "unknown"));
```

Comment:
Minor but my brain likes the ordering of the test to be the same as the ordering in the logic. In SERVE we typically evaluate things "from small to big", so hotspots -> codons -> exons -> genes -> fusions.